components/map/map.service.js

```diff
@@ -517,9 +517,10 @@ export default function (
     const existingUrls = angular.isUndefined(existingSource.getUrls)
       ? ''
       : existingSource.getUrls();
-    const newUrls = angular.isUndefined(newSource.getUrls)
+    let newUrls = angular.isUndefined(newSource.getUrls)
       ? ['']
       : newSource.getUrls();
+    newUrls = newUrls ? newUrls : [''];
```

Comment:
You are saving boolean or array into this variable. Seems confusing. Didn't you want to save the actual content of `newSource.getUrls` instead of `true`/`false`?

Reply:
Do I ? Yes exactly, wanted to chcek if its defined basically and assign a value if not